csv/src/main/java/com/hazelcast/jet/csv/CsvSource.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.csv;
+
+import com.hazelcast.jet.Traverser;
+import com.hazelcast.jet.Traversers;
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.pipeline.BatchSource;
+import org.simpleflatmapper.csv.CsvMapper;
+import org.simpleflatmapper.csv.CsvMapperFactory;
+
+import javax.annotation.Nonnull;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import static com.hazelcast.jet.Traversers.traverseStream;
+import static com.hazelcast.jet.pipeline.Sources.batchFromProcessor;
+
+/**
+ * Reads CSV files and maps lines to Java Beans.
+ *
+ * Please note, that Java Bean should be a public class, in order to work correctly with SimpleFlatMapper.
+ */
+public final class CsvSource<R> extends AbstractProcessor {
```

Comment:
would be better to rename this ReadCsvP and have another utility class called CsvSources that contains the actual source